libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1047,15 +1045,31 @@ string YulUtilFunctions::storageArrayPushZeroFunction(ArrayType const& _type)
 				if iszero(lt(oldLen, <maxArrayLength>)) { <panic>() }
 				sstore(array, add(oldLen, 1))
```

Comment:
Can you try to remove everything from this function after writing the length here? Since we should take care that storage is always zeroed, we should be fine without doing anything besides writing the new length, shouldn't we? We can see if any tests fail that way.

Reply:
I agree to remove everything and I did it :) Hopefully all the tests will pass